src/backend/distributed/executor/executor.c

```diff
@@ -1524,12 +1525,18 @@ ManageWorkerPool(WorkerPool *workerPool)
 	/* we should never have less than 0 connections ever */
 	Assert(activeConnectionCount >= 0 && idleConnectionCount >= 0);
 
+	if (workerPool->failed)
+	{
+		/* connection pool failed */
+		return;
+	}
+
 	/* we might fail the execution or warn the user about connection timeouts */
 	CheckConnectionTimeout(workerPool);
 
 	if (failedConnectionCount >= 1)
```

Comment:
shall we move this check above
```C

if (workerPool->failed || failedConnectionCount >= 1)
{
    return
}
```

Reply:
I figured there might be situations in which we want to check for timeout after one connection has failed, e.g. we open 4 connections and 1 fails and the second never responds.